### DISCUSSION TITLE
Integration with Spring boot.

### DISCUSSION THREAD

**Discussion by ArTemmey** [Foso/Ktorfit#49] [Ideas]
Ktorfit API is very similar to the Spring boot REST controllers API. Wouldn't it be a good idea to combine these two APIs so that we can create a server and access it using the same interface? For example, you can configure your annotation processor to process Spring annotations.

**Comment by Foso** [comment]
Hi, i'm not very familiar with the Spring boot and the REST controllers API. I can take a look at it and see if it's a doable task.

**Reply by ArTemmey** [reply]
@Foso it would be great, thanks! 

The fact is that although the backend and frontend/mobile interact closely, they are two separate systems. Because of this, their interaction is difficult to organize (how we organize the interaction of two components in one program). At best, Swagger is used for this, and at worst, it’s just documentation or even a verbal agreement. But even Swagger has its own difficulties and requires additional competencies and more development time.

Spring boot is one of the more popular server ecosystems and is powered by Java/Kotlin. And thanks to Kotlin Multiplatform, it is possible to combine the server and frontend/mobile into one inseparable codebase. Where an API contract is a regular Kotlin interface that we call on the frontend and implement on the backend. Thus, we solve many problems of integrating backend and frontend. I think many developers would be very happy with such a solution.
